core/dive.h

```diff
@@ -290,7 +290,7 @@ struct divecomputer {
 	struct divecomputer *next;
 };
 
-#define MAX_CYLINDERS (8)
+#define MAX_CYLINDERS (20)
```

Comment:
That doesn't work. Well, it sorta does right now, but it doesn't solve your problem.
You have now made the data structure bigger, but you now also allow 20 tanks in both a regular dive and a planned dive.
I would leave that at 8 tanks (because beyond that, the UI becomes unusable, anyway) and simply warn the user when merging a dive and planned dive exceeds 8 tanks and then drop the remaining tanks.